projectq/backends/_azure/_azure_quantum.py

```diff
@@ -0,0 +1,403 @@
+# -*- coding: utf-8 -*-
+#   Copyright 2021 ProjectQ-Framework (www.projectq.ch)
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Back-end to run quantum programs using Azure Quantum."""
+
+import numpy as np
+from collections import Counter
+
+from projectq.types import WeakQubitRef
+from projectq.cengines import BasicEngine
+from projectq.meta import LogicalQubitIDTag
+from projectq.ops import (
+    AllocateQubitGate,
+    DeallocateQubitGate,
+    FlushGate,
+    MeasureGate
+)
+
+from ._azure_quantum_client import send, retrieve
+from ._util import (
+    IONQ_PROVIDER_ID,
+    QUANTINUUM_PROVIDER_ID,
+    is_available_ionq,
+    is_available_quantinuum,
+    to_json,
+    to_qasm,
+    rearrange_result
+)
+
+from ._exceptions import AzureQuantumTargetNotFoundError
+
+try:
+    from azure.quantum import Workspace
+    from azure.quantum.target import Target, IonQ, Quantinuum
+    from azure.quantum.target.target_factory import TargetFactory
+except ImportError:  # pragma: no cover
+    raise ImportError(
+        "Missing optional 'azure-quantum' dependencies. To install run: pip install projectq[azure-quantum]"
+    )
```

Comment:
This will not work. While the error message is informative, the way it is currently implemented will always trigger an error when people try to import `projectq.backends` when the `azure-quantum` dependency is not installed.

You need to change the code to trigger the error only when the users *actually* try to use the `AzureQuantumBackend`.

Reply:
Thanks for pointing this out, I updated the code to avoid this scenario. Please take a look.